src/giskardpy/goals/align_to_push_door.py

```diff
@@ -0,0 +1,106 @@
+                 root_link: str,
+                 tip_link: str,
+                 door_object: str,
+                 door_height: float,
+                 door_length: float,
+                 tip_gripper_axis: Vector3Stamped,
+                 root_group: Optional[str] = None,
+                 tip_group: Optional[str] = None,
+                 reference_linear_velocity: float = 0.1,
+                 reference_angular_velocity: float = 0.5,
+                 weight: float = WEIGHT_BELOW_CA,
+                 name: Optional[str] = None,
+                 start_condition: cas.Expression = cas.TrueSymbol,
+                 hold_condition: cas.Expression = cas.FalseSymbol,
+                 end_condition: cas.Expression = cas.TrueSymbol):
+        """
+        The objective is to reach an intermediate point before pushing the door
+        """
+        # ToDo: How to get the joint axis
+        self.root = god_map.world.search_for_link_name(root_link, root_group)
+        self.tip = god_map.world.search_for_link_name(tip_link, tip_group)
+        self.door_object = god_map.world.search_for_link_name(door_object)
+        self.door_length = door_length
+        self.door_height = door_height
+        self.reference_linear_velocity = reference_linear_velocity
+        self.reference_angular_velocity = reference_angular_velocity
+        self.weight = weight
+
+        object_joint_name = god_map.world.get_movable_parent_joint(self.door_object)
+        object_joint_angle = god_map.world.state[object_joint_name].position
+
+        tip_gripper_axis.header.frame_id = self.tip
+        tip_gripper_axis.vector = tf.normalize(tip_gripper_axis.vector)
+        self.tip_gripper_axis = tip_gripper_axis
+        object_V_object_rotation_axis = cas.Vector3(god_map.world.get_joint(object_joint_name).axis)
+
+        if name is None:
+            name = f'{self.__class__.__name__}/{self.root}/{self.tip}'
+        super().__init__(name=name)
+
+        root_T_tip = god_map.world.compose_fk_expression(self.root, self.tip)
+        root_T_door_expr = god_map.world.compose_fk_expression(self.root, self.door_object)
+        tip_V_tip_grasp_axis = cas.Vector3(self.tip_gripper_axis)
+        root_V_object_rotation_axis = cas.dot(root_T_door_expr, object_V_object_rotation_axis)
+        root_V_tip_grasp_axis = cas.dot(root_T_tip, tip_V_tip_grasp_axis)
+        root_T_door = god_map.world.compute_fk_pose(self.root, self.door_object)
+        door_T_root = god_map.world.compute_fk_pose(self.door_object, self.root)
+
+        root_P_intermediate_point = cas.Point3(root_T_door.pose.position)
+        if root_V_object_rotation_axis.y == 1:
+            # 3/4 of the height as the tip has to be a little further inside the object
+            root_P_intermediate_point[2] = root_P_intermediate_point[2] + self.door_height*3/4
+        elif root_V_object_rotation_axis.z == 1:
+            root_P_intermediate_point[1] = root_P_intermediate_point[1] + self.door_length*3/4
```

Comment:
what if x=1?

Reply:
that would usually be a drawer that moves along x